cob_twist_controller/include/cob_twist_controller/constraints/constraint_impl.h

```diff
@@ -0,0 +1,345 @@
+/*!
+ *****************************************************************
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2015 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_twist_controller
+ *
+ * \author
+ *   Author: Marco Bezzon, email: Marco.Bezzon@ipa.fraunhofer.de
+ *
+ * \date Date of creation: May, 2015
+ *
+ * \brief
+ *   Implementation of several constraints
+ *
+ ****************************************************************/
+
+#ifndef CONSTRAINT_IMPL_H_
+#define CONSTRAINT_IMPL_H_
+
+#include <boost/pointer_cast.hpp>
+
+#include <ros/ros.h>
+
+#include <kdl/chainiksolver.hpp>
+#include <kdl/chainjnttojacsolver.hpp>
+#include <kdl/jntarray.hpp>
+
+#include "cob_twist_controller/constraints/constraint.h"
+#include "cob_twist_controller/constraints/constraint_params.h"
+
+/* BEGIN ConstraintsBuilder *************************************************************************************/
+/**
+ * Static builder method to create damping methods dependent on parameterization.
+ */
+template <typename PRIO>
+std::set<tConstraintBase> ConstraintsBuilder<PRIO>::createConstraints(InvDiffKinSolverParams &inv_diff_kin_solver_params,
+                                                                       const KDL::JntArray& q,
+                                                                       const Matrix6Xd &jacobian_data,
+                                                                       KDL::ChainJntToJacSolver& jnt_to_jac,
+                                                                       CallbackDataMediator& data_mediator)
+{
+    std::set<tConstraintBase> constraints;
+    if (GPM_JLA == inv_diff_kin_solver_params.constraint)
+    {
+        typedef JointLimitAvoidance<ConstraintParamsJLA, PRIO> tJla;
+        ConstraintParamsJLA params = ConstraintParamFactory<ConstraintParamsJLA>::createConstraintParams(inv_diff_kin_solver_params, data_mediator);
+        // TODO: take care PRIO could be of different type than UINT32
+        boost::shared_ptr<tJla > jla(new tJla(100, q, params));
```

Comment:
For the current version, because we do not have a priority management at the moment.